File: episoq.js
javascript: (() => {
  const a = location.href.split('/');
  if(a[2] != 'scrapbox.io') return;
  const project = a[3];
  const page = a[4];
  const url = `https://scrapbox.io/api/code/${project}/${page}/episopass.json`;
  $.ajax({
    url: url,
    async: false,
    success: (data) => {
      questions = data.questions;
      n = questions.length;
      while (n) {
        const i = Math.floor(Math.random() * n--);
        [questions[i], questions[n]] = [questions[n], questions[i]]
      }
      w = window.open();
      w.location.href = `http://EpisoPass.com/easy.html?questions=${data.questions.slice(0,10).join(';')}&answers=${data.answers.join(';')}`
    }
  });
})();


      
      w = window.open();
      w.document.write("<pre>"+data.questions.slice(0,10).join("\n")+"</pre>");
      w.document.write("<pre>"+data.answers.join("\n")+"</pre>");

File: episoq.rb
#!/usr/bin/env ruby
# coding: utf-8
#
# Scrapbox上にリストしたなぞなぞ問題プールからランダムに問題を選択して問題編集画面を開く
# https://github.com/masui/EpisoQ/ を利用する
#
# データ例: https://Scrapbox.io/masui/県名なぞなぞ
#
# % episoq masui 県名なぞなぞ 5
#
project = ARGV[0]    || 'masui'
pagetitle = ARGV[1]  || '県名なぞなぞ'
nquestions = ARGV[2] || 10
js = ARGV[3]         || 'episopass.js'

url = "https://masui.github.io/EpisoQ/?qa=https://scrapbox.io/api/code/#{project}/#{pagetitle}/#{js}&n=#{nquestions}"

open = "open"
open = "xdg-open" if `which xdg-open` != ''
system "#{open} '#{url}'"
